Index: llvm-toolchain-3.8-3.8.1/lldb/cmake/modules/LLDBConfig.cmake
===================================================================
--- llvm-toolchain-3.8-3.8.1.orig/lldb/cmake/modules/LLDBConfig.cmake
+++ llvm-toolchain-3.8-3.8.1/lldb/cmake/modules/LLDBConfig.cmake
@@ -165,13 +165,6 @@ function(find_python_libs_windows)
 endfunction(find_python_libs_windows)
 
 if (NOT LLDB_DISABLE_PYTHON)
-  if(UNIX)
-    # This is necessary for crosscompile on Ubuntu 14.04 64bit. Need a proper fix.
-    if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-      set(CMAKE_LIBRARY_ARCHITECTURE "x86_64-linux-gnu")
-    endif()
-  endif()
-
   if ("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
     find_python_libs_windows()
 
