Index: b/polly/lib/Makefile
===================================================================
--- a/polly/lib/Makefile
+++ b/polly/lib/Makefile
@@ -26,9 +26,9 @@
                    CodeGen/IslNodeBuilder.cpp \
                    CodeGen/CodeGeneration.cpp
 
-POLLY_JSON_FILES= JSON/json_reader.cpp \
-                  JSON/json_value.cpp \
-                  JSON/json_writer.cpp
+#POLLY_JSON_FILES= JSON/json_reader.cpp \
+#                  JSON/json_value.cpp \
+#                  JSON/json_writer.cpp
 
 ISL_FILES= External/isl/basis_reduction_tab.c \
            External/isl/isl_aff.c \
@@ -146,4 +146,5 @@
 #
 include $(LEVEL)/Makefile.common
 
-LIBS += $(POLLY_LD) $(POLLY_LIB)
+LIBS += $(POLLY_LD) $(POLLY_LIB) -ljsoncpp
+
Index: b/polly/lib/CMakeLists.txt
===================================================================
--- a/polly/lib/CMakeLists.txt
+++ b/polly/lib/CMakeLists.txt
@@ -1,10 +1,10 @@
 set(LLVM_NO_RTTI 1)
 
-set(POLLY_JSON_FILES
-    JSON/json_reader.cpp
-    JSON/json_value.cpp
-    JSON/json_writer.cpp
-)
+# set(POLLY_JSON_FILES
+#     JSON/json_reader.cpp
+#     JSON/json_value.cpp
+#     JSON/json_writer.cpp
+# )
 
 set(ISL_CODEGEN_FILES
     CodeGen/IslAst.cpp
@@ -46,7 +46,7 @@
   Support/RegisterPasses.cpp
   Support/ScopHelper.cpp
   Support/ScopLocation.cpp
-  ${POLLY_JSON_FILES}
+#  ${POLLY_JSON_FILES}
   Transform/Canonicalization.cpp
   Transform/CodePreparation.cpp
   Transform/DeadCodeElimination.cpp
@@ -54,7 +54,7 @@
   ${POLLY_HEADER_FILES}
   )
 
-target_link_libraries(Polly PollyISL)
+target_link_libraries(Polly PollyISL jsoncpp)
 
 if (BUILD_SHARED_LIBS)
   target_link_libraries(Polly
Index: b/polly/CMakeLists.txt
===================================================================
--- a/polly/CMakeLists.txt
+++ b/polly/CMakeLists.txt
@@ -128,10 +128,13 @@
   INCLUDE_DIRECTORIES( ${CUDALIB_INCLUDE_DIR} )
 endif(CUDALIB_FOUND)
 
+find_path(JSONCPP_INCLUDE_PATH json/json.h PATH_SUFFIXES jsoncpp)
+
 include_directories(
   BEFORE
   ${CMAKE_CURRENT_SOURCE_DIR}/include
-  ${CMAKE_CURRENT_SOURCE_DIR}/lib/JSON/include
+#  ${CMAKE_CURRENT_SOURCE_DIR}/lib/JSON/include
+  ${JSONCPP_INCLUDE_PATH}
   ${CMAKE_CURRENT_BINARY_DIR}/lib/External/isl/include
   ${CMAKE_CURRENT_SOURCE_DIR}/lib/External/isl/include
   ${CMAKE_CURRENT_BINARY_DIR}/include
Index: b/polly/Makefile.config.in
===================================================================
--- a/polly/Makefile.config.in
+++ b/polly/Makefile.config.in
@@ -53,7 +53,7 @@
 POLLY_INC :=  @cuda_inc@ \
               -I$(POLLY_OBJ_ROOT)/lib/External/isl/include \
               -I$(POLLY_OBJ_ROOT)/lib/External/isl \
-              -I$(POLLY_SRC_ROOT)/lib/JSON/include \
+              -I/usr/include/jsoncpp \
               -I$(POLLY_SRC_ROOT)/lib/External/isl/include \
               -I$(POLLY_SRC_ROOT)/lib/External/isl/imath \
               -I$(POLLY_SRC_ROOT)/lib/External/isl
